src/utils/ManPages.ts

```diff
@@ -0,0 +1,98 @@
+import {execFile} from "child_process";
+import {Suggestion, styles, unique} from "../plugins/autocompletion_providers/Common";
+import {AutocompletionProvider} from "../Interfaces";
+
+const combineManPageLines = (lines: string[]) => lines
+    .map(line => line.trim())
+    .reduce(
+        (memo, next) => {
+            if (next.endsWith("-")) {
+                return memo.concat(next.slice(0, -1));
+            } else {
+                return memo.concat(next, " ");
+            }
+        },
+        ""
+    );
+
+// Note: this is still pretty experimental. If you want to do man page parsing
+// for a new command, expect to have to make some changes here.
+export let manPageToOptions = (command: string): Promise<AutocompletionProvider> => {
+    // use execFile to prevent a command like "; echo test" from running the "echo test"
+    return new Promise((resolve, reject) => {
```

Comment:
Please use `async/await` instead of promises. You'd need to create a promisified wrapper for `execFile` in `utils/Common.ts`. There are many such wrappers to copy from.